tests/openassetio/hostAPI/test_transactions.py

```diff
@@ -242,23 +319,50 @@ def test_scope_does_not_cancel(self, a_scoped_group):
 
         with pytest.raises(RuntimeError):
             with a_scoped_group:
-                self.__assertActionGroupCalls(
-                    mock_coordinator, push=a_context, pop=None, cancel=None)
                 raise RuntimeError
 
-            self.__assertActionGroupCalls(mock_coordinator, push=None, pop=a_context, cancel=None)
+            assert_action_group_calls(mock_coordinator, push=None, pop=a_context, cancel=None)
 
-    @staticmethod
-    def __assertActionGroupCalls(mock_coordinator, push, pop, cancel):
 
-        for method, arg in (
-                (mock_coordinator.pushActionGroup, push),
-                (mock_coordinator.popActionGroup, pop),
-                (mock_coordinator.cancelActions, cancel)
-        ):
-            if arg is None:
-                method.assert_not_called()
-            else:
-                method.assert_called_once_with(arg)
+#
+# Utilities
+#
 
-        mock_coordinator.reset_mock()
+def assert_action_group_calls(mock_coordinator, push, pop, cancel):
+    """
+    A convenience methods that checks the actionGroup related methods
+    of the supplied mock TransactionCoordinator object have been called
+    with the specified arguments. If any of the supplied arguments are None,
+    then it will be asserted that the method was not called.
+    """
+
+    for method, arg in (
+            (mock_coordinator.pushActionGroup, push),
+            (mock_coordinator.popActionGroup, pop),
+            (mock_coordinator.cancelActions, cancel)
+    ):
+        if arg is None:
+            method.assert_not_called()
+        else:
+            method.assert_called_once_with(arg)
+
+    mock_coordinator.reset_mock()
+
+def assert_transaction_calls(mock_manager, mock_host_session, start, finish, cancel):
```

Comment:
[consider][minor] Just to point out a handy trick with pytest fixtures, whereby you can return a function.  

So in this case  you could reduce the number of arguments in test cases by something like

```
@pytest.fixture
def assert_transaction_calls(mock_manager, mock_host_session):
    def fn(start, finish, cancel):
        ....
    return fn
```
then the test cases only need to fill in the `start`, `finish` and `cancel` arguments.

Very minor though, probably not worth the effort to change now, but a useful trick to bear in mind.

Reply:
I left this one as not all the arguments are supplied by other fixtures, but good to know, thanks!